packages/nitro-protocol/contracts/AssetHolder.sol

```diff
@@ -264,11 +283,45 @@ contract AssetHolder is IAssetHolder {
                 )
             );
         } else {
-            delete outcomeHashes[guarantee.targetChannelId];
+            delete assetOutcomeHashes[guarantee.targetChannelId];
         }
 
     }
 
+    struct FunctionCall {
+        bool isClaimAll; // otherwise transferAll implied
```

Comment:
This could be an enum `FunctionType`.

Reply:
I'm bouncing this generalised functionality to future work. This PR is now focussed only on convenience methods that manage a single channel.